src/Core/System/Snippet/Command/Util/CountryAgnosticFileValidator.php

```diff
@@ -0,0 +1,175 @@
+
+                if ($isApp) {
+                    $accumulator[] = $appPath;
+                }
+
+                if ($isPlugin) {
+                    $accumulator[] = $pluginPath;
+                }
+
+                return $accumulator;
+            }, []);
+
+            $finder->in($extensionPaths);
+        }
+
+        return $finder;
+    }
+
+    public function checkTranslationFiles(ValidatedTranslationFileOptions $options): ValidatedTranslationFileStruct
+    {
+        $finder = $this->getFinder($options);
+        if ($finder->count() < 0) {
+            return new ValidatedTranslationFileStruct();
+        }
+
+        $languageFiles = new TranslationFileCollection([]);
+        $countrySpecificFileCollection = new TranslationFileCollection([]);
+        foreach ($finder as $file) {
+            $filename = $file->getFilename();
+
+            $isCoreTranslationFile = preg_match(
+                SnippetPatterns::CORE_SNIPPET_FILE_PATTERN,
+                $filename,
+                $coreFileData,
+                \PREG_UNMATCHED_AS_NULL
+            );
+
+            $isAdminTranslationFile = preg_match(
+                SnippetPatterns::ADMIN_SNIPPET_FILE_PATTERN,
+                $filename,
+                $adminFileData,
+                \PREG_UNMATCHED_AS_NULL
+            );
+
+            if (!$isAdminTranslationFile && !$isCoreTranslationFile) {
+                continue;
+            }
+
+            $currentFileData = $isAdminTranslationFile ? $adminFileData : $coreFileData;
+
+            $currentDomain = $currentFileData['domain'] ?? 'administration';
+            $formatedLanguageStruct = new TranslationFile(
+                $filename,
+                $file->getPath(),
+                $currentDomain,
+                str_replace('_', '-', $currentFileData['locale']),
+                $currentFileData['language'] ?? null,
+                $currentFileData['script'] ?: null,
+                $currentFileData['region'] ?: null,
+                !$isAdminTranslationFile && !empty($currentFileData['isBase']),
+            );
+
+            $fileIdentifier = $formatedLanguageStruct->getFullPath();
+
+            if ($formatedLanguageStruct->getRegion()) {
+                $countrySpecificFileCollection->set($fileIdentifier, $formatedLanguageStruct);
+            }
+
+            $languageFiles->set($fileIdentifier, $formatedLanguageStruct);
+        }
+
+        return $this->processAgnosticFiles(new ValidatedTranslationFileStruct(
+            $languageFiles,
+            $countrySpecificFileCollection,
+        ));
+    }
+
+    public function processAgnosticFiles(ValidatedTranslationFileStruct $validatedFileStruct): ValidatedTranslationFileStruct
+    {
+        $count = $validatedFileStruct->getSpecificCollection()->count();
+        if ($count < 1) {
+            return $validatedFileStruct;
+        }
+
+        foreach ($validatedFileStruct->getSpecificCollection()->getElements() as $countrySpecificFile) {
```

Comment:
```suggestion
        foreach ($validatedFileStruct->getSpecificCollection() as $countrySpecificFile) {
```

calling `getElements` explicitly is not needed